test/service/IconService.test.js

```diff
@@ -0,0 +1,231 @@
+import { IconResult, IconService } from '../../src/services/IconService';
+import { loadBvvIcons } from '../../src/services/provider/icons.provider';
+import { $injector } from '../../src/injection';
+
+
+describe('IconsService', () => {
+
+	const configServiceMock = {
+		getValue: () => { },
+		getValueAsPath: () => { }
+	};
+
+	beforeAll(() => {
+		$injector
+			.registerSingleton('ConfigService', configServiceMock);
+	});
+
+	const getMatcher = (id) => {
+		return (idOrUrl) => idOrUrl === id || idOrUrl.endsWith(id);
+	};
+	const iconResult1 = new IconResult('foo1', 'bar1', getMatcher('foo1'), () => 'http://some.url/foo1');
+	const iconResult2 = new IconResult('foo2', 'bar2', getMatcher('foo2'), () => 'http://some.url/foo2');
+	const markerIconResult = new IconResult('marker', 'marker', getMatcher('marker'), () => 'http://some.url/marker');
+	const loadMockIcons = async () => {
+		return [
+			iconResult1,
+			iconResult2
+		];
+	};
+
+
+	const setup = (iconProvider = loadMockIcons) => {
+		return new IconService(iconProvider);
+	};
+
+	describe('initialization', () => {
+
+		it('initializes the service', async () => {
+			const instanceUnderTest = setup();
+			expect(instanceUnderTest._icons).toBeNull();
+
+			const icons = await instanceUnderTest.all();
+			const defaultIcon = instanceUnderTest.getDefault();
+
+			expect(icons.length).toBe(3);
+			expect(icons[0]).toEqual(defaultIcon);
+		});
+
+		it('initializes the service with default provider', async () => {
+			const instanceUnderTest = new IconService();
+			expect(instanceUnderTest._iconProvider).toEqual(loadBvvIcons);
```

Comment:
no test for default  `_iconColorProvider`

Reply:
Thanks